scripts/core/CONTRIB.rst

```diff
@@ -216,6 +216,14 @@ implement your experimental feature.
     $ python scripts/add_experimental_feature.py <name-of-your-experimental-feature>
 
 
+Experimental features *must* not make any changes to the core YaML files and 
+*must* be described entirely in their own YaML file. Sometimes, however 
```

Comment:
Are there deffo no cases now where core yaml files would need to be touched?

Reply:
Just want to ensure these uses of *must* are not too strong.

Reply:
Extending existing structs/unions (e.g., adding a new partition method)?. But experimental features should probably avoid doing things like that in the first place.

Reply:
Yeah I think extending existing structs is not allowed for experimental features. Hopefully if this is needed it can be done with the `pNext` chain. Unions I'm less sure about.

Reply:
Extending a struct/union implies an ABI change which I think we should avoid for experimental features. That's how I've been thinking about this.